solarforecastarbiter/reference_forecasts/utils.py

```diff
@@ -1,45 +1,50 @@
 import numpy as np
 import pandas as pd
+import pytz
 
 
 from solarforecastarbiter.io import utils as io_utils
 
 
-def get_issue_times(forecast, start_from=None):
-    """
-    Return a list of the issue times for a given Forecast.
+def get_issue_times(forecast, start_from):
+    """Return a list of the issue times for a given Forecast starting
+    from the date of start_from until the first issue time of the  next day.
+    The output timestamps are localized to the timezone of `start_from`.
 
     Parameters
     ----------
     forecast : datamodel.Forecast
         Forecast object that contains the time information
-    start_from : pandas.Timestamp or None
-        If a timestamp, return the issues times for the same days as
-        `start_from`. If None, return datetime.time objects.
+    start_from : pandas.Timestamp
+        Return issue times from this same day in the same timezone
 
     Returns
     -------
     list
-        Either of datetime.time objects indicating the possible issue times, or
         pandas.Timestamp objects with the issues times for the particular day
         including the first issue time for the next day.
     """
-    if start_from is None:
-        issue = pd.Timestamp.combine(pd.Timestamp(0).date(),
-                                     forecast.issue_time_of_day)
-    else:
-        issue = pd.Timestamp.combine(start_from.date(),
-                                     forecast.issue_time_of_day).tz_localize(
-                                         start_from.tz)
-    next_day = (issue + pd.Timedelta('1d')).floor('1d')
-    # works even for midnight issue
-    out = [issue]
-    while issue < next_day:
-        issue += forecast.run_length
-        out.append(issue)
-    if start_from is None:
-        out = [o.time() for o in out]
-    return out
+    start_time = forecast.issue_time_of_day
+    if start_time.tzinfo is None:
+        start_time = pytz.utc.localize(start_time)
+    # work for forecasts over 1d run length
+    dayadj = pd.Timedelta(forecast.run_length).ceil('1d')
+    earliest_start = pd.Timestamp.combine(
+        (start_from - dayadj).date(), start_time)
+    possible_times = []
+    for i in range(3):
+        start = earliest_start + i * dayadj
+        end = (start + dayadj).floor('1d')
+        possible_times.extend(list(
+            pd.date_range(start=start, end=end, freq=forecast.run_length)))
+    # broad range of times that should cover start_from and next time
+    # even after timezone conversion
```

Comment:
the lines above made more sense once I got to this comment. I don't understand the motivation for this approach though. It's also not clear that this works if `forecast.run_length` is very short.

Reply:
for example if issue time is 06:00 UTC and start = 20200101T0000Z with a run_length of 1 hour, 2020-01-01T00:00Z is a valid start time that should be returned. As it's written we need to include the previous day (line 33) to generate that issue time. We also need to get the issue times for the next day to get the first one (2020-01-02T06:00Z), hence  i in range(3) (yesterday, today, tomorrow). It is certainly too broad, but all the edge cases are handled. 

The tests show that it does work for short run lengths (many are 1hour, shorter would be fine too) because of the `ceil` on line 31.